### PR TITLE
Refactor tests related to notBefore and nbf (#492)

### PR DESCRIPTION
**Pull request by MitMaro**
First PR for #492.

This change extracts all tests in the current files related to `notBefore` and `nbf` into a single test file. It also adds several missing related tests.

I believe I removed all references to `nbf` and `notBefore` from the current tests.

~~There are two TODOs in the tests, because I found a few unhandled cases that I expected would cause an error but did not. They are:~~

* ~~Passing an empty string to `notBefore`. I'm not sure if this should default to `0` or should throw an error. Currently it throws a `val is not a non-empty string or a valid number. val=""` from `ms`.~~
* ~~Passing `-Infinity`, `Infinity` and `NaN` to `nbf`. These all `JSON.stringify` to `null`, so I would assume should be considered invalid.~~

Any suggestions of missing tests are more than welcome, and I will gladly add!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,8 @@
   "description": "JSON Web Token implementation (symmetric and asymmetric)",
   "main": "index.js",
   "scripts": {
-    "test": "nyc --reporter=html --reporter=text mocha && nsp check && cost-of-modules"
+    "test:unit": "nyc --reporter=html --reporter=text mocha",
```
Comment (ziluvatar) [inline package.json]: 🤔 they are not unit tests. I guess you wanted a fast way to develop the tests getting coverage feedback without the delay of nsp + cost-of-modules, right? since this is test + coverage, let's name it: `coverage` or `test:coverage` or similar.

File: test/nbf.test.js [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+      true,
+      false,
+      null,
+      -1.1,
+      1.1,
+      -Infinity,
+      Infinity,
+      NaN,
+      // TODO empty string currently fails
+      // '',
```
Comment (ziluvatar) [inline test/nbf.test.js]: Let's create a test with the current behavior and a PR to fix this, since it should be invalid too (with same error message), and we can merge the fix on next major release.

File: test/nbf.test.js [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+  describe('`jwt.sign` nbf claim validation', function () {
+    [
+      true,
+      false,
+      null,
+      undefined,
+      // TODO should these fail in validation?
+      // -Infinity,
+      // Infinity,
+      // NaN,
```
Comment (ziluvatar) [inline test/nbf.test.js]: As you said, yes, I think they should fail, like I said before, let's add the current behavior and a PR (same one as before?) to fix in next major.

File: package.json (hunk 1 above)
Comment (MitMaro) [reply package.json]: Sure thing! `cost-of-modules` temporarily renames the `node_modules/` directory, and it was causing my editor to continually re-index. This was a way to run the tests without those checks.

I'm okay with a `coverage` task.

File: test/nbf.test.js (hunk 2 above)
Comment (MitMaro) [reply test/nbf.test.js]: Just to make sure I understand correctly before I update:

* add a test for an empty string that checks for the `ms` error;
* add another PR that fixes the behaviour

Comment (MitMaro) [reply test/nbf.test.js]: Fixed in https://github.com/MitMaro/node-jsonwebtoken/commit/b037c26 before I squashed commits.

File: test/nbf.test.js (hunk 3 above)
Comment (MitMaro) [reply test/nbf.test.js]: Fixed in https://github.com/MitMaro/node-jsonwebtoken/commit/b037c26 before I squashed commits.

**Comment by MitMaro** [comment]
I fixed some of the formatting in https://github.com/MitMaro/node-jsonwebtoken/commit/943dab9 after fixing the issues noted above.

File: test/nbf.test.js (hunk 2 above)
Comment (ziluvatar) [reply test/nbf.test.js]: Yes
